### PR TITLE
Merging ModelTrainer and TrainingAdapter class

### PR DESCRIPTION
**Pull request by wiederm**
## Description
This  PR combined the two separate training classes. There wasn't a good incentive to keep them separated and combining them will enable cleaner interfaces and less redundant code.

## Todos
  - [x] Combine classes
  - [x] Update tests
  - [x] Update docstrings/documentation

## Questions
- [ ] Question1

## Status
- [x] Ready to go

### CODE REVIEW

File: modelforge/potential/ani.py [hunk 1]
```diff
@@ -664,6 +664,7 @@ def __init__(
         activation_function_parameter: Dict,
         postprocessing_parameter: Dict[str, Dict[str, bool]],
         dataset_statistic: Optional[Dict[str, float]] = None,
+        model_seed: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/ani.py:667]: we should also probably call this `potential_seed` for consistency.

File: modelforge/potential/models.py [hunk 2]
```diff
@@ -971,6 +996,11 @@ def __init__(
         super().__init__()
         from modelforge.utils.units import _convert_str_to_unit
 
+        if model_seed:
+            import torch
+
+            torch.manual_seed(model_seed)
```
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/models.py:1002]: I'll note that, since this doesn't seem to really do anything at the moment, we could probably hold off on adding this as a user modifiable parameter (i.e., sticking it in the potential pydantic model)

File: modelforge/potential/ani.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/ani.py:667]: Agreed, `potential_seed` is the better naming, since we started to move to calling it `potential` instead of `model`.

File: modelforge/potential/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/models.py:1002]: This change was motivated by a failure in this [test](https://github.com/choderalab/modelforge/blob/a4b6cda0a9a340b95c7a2df868147a22d4eaa73e/modelforge/tests/test_models.py#L481) at this assert [statment](https://github.com/choderalab/modelforge/blob/a4b6cda0a9a340b95c7a2df868147a22d4eaa73e/modelforge/tests/test_models.py#L531).

In this test, we were setting the seed manually for each `model_inference = NeuralNetworkPotentialFactory.generate_potential()` method call. But this resulted in failures for two potentials (can't remember which). I agree that this might not be the best motivation for this change, and it is only useful for tests, but that's also the reason why I wouldn't expose it in the TOML file and keep the default as `None`.

File: modelforge/potential/models.py [hunk 3]
```diff
@@ -540,19 +540,41 @@ def apply_bwd(res, grads):
...
-        model_parameter: Dict[str, Union[str, Any]],
+        potential_parameter: Union[
+            ANI2xParameters,
+            SAKEParameters,
+            SchNetParameters,
+            PhysNetParameters,
+            PaiNNParameters,
+            TensorNetParameters,
+        ],
         simulation_environment: Literal["PyTorch", "JAX"] = "PyTorch",
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/potential/models.py:572]: Agreed, we can remove the defaults here and control it via the runtime config

I can access this attribute in the initialized Pydantic model and set it to a different value at runtime (only necessary for testing), right?

File: modelforge/tests/test_models.py [hunk 4]
```diff
@@ -487,14 +491,31 @@ def test_calculate_energies_and_forces(potential_name, single_batch_with_batchsi
...
+        E_training.sum(), nnp_input.positions, create_graph=True, retain_graph=True
+    )[0]
+
+    # compare to inference model
+    model_inference = NeuralNetworkPotentialFactory.generate_potential(
         use="inference",
-        model_parameter=config["potential"].model_dump(),
+        potential_parameter=config["potential"],
+        dataset_statistic=model_training.dataset_statistic,
+        model_seed=42,
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/tests/test_models.py:514]: As mentioned above, I think this is only useful for testing. 

I think the scenario that will come up is that we need to make sure that we can reproduce the random number stream. That includes, as you mentioned, **all** decisions based on a random number. In that case it makes sense to set this much earlier, ideally at initialization of the `TrainModel` (which we should also call `TrainPotential`) class.

[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/tests/test_models.py:514]: But, to that point, I think we should set this in the TOML file (I'd recommend setting it in the runtime TOML) and set the seed in the Trainer class.

File: modelforge/potential/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/models.py]: Sounds reasonable.  This should be ultimately used by PyTorch lightning, so good to have the option to set.

File: modelforge/potential/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/models.py]: I initially removed it, then realized we use this in tests.  I changed the code so that it would use the environment from the runtime_parameters (if provided), which it wasn't doing.  If we set simulation_environment explicitly, it will override it.  If not provided in either case, it defaults to PyTorch.  Again, I think mostly just useful for testing.

**Review by chrisiacovella — APPROVED** [review]
